controllers/chat/getChannelDetail.js

```diff
@@ -59,7 +60,15 @@ const getChannelDetail = async (req, res) => {
     better_channel_members: betterChannelMember,
     better_channel_member_objects: betterChannelMemberObject,
     ...updatedChannel,
-    messages: messages?.results?.map((message) => message?.message)
+    messages: messages?.results?.map((message) => {
+      let detail_message = message?.message;
+
+      if (detail_message.message_type === MESSAGE_TYPE.TEXT) {
+        detail_message.attachments = [];
+      }
+
+      return detail_message;
+    })
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change introduces a potential issue. The `message` object is being directly modified, which could lead to unintended side effects if this object is used elsewhere in the code. Instead of modifying the `message` object directly, consider creating a copy of it before making changes.

```diff
- let detail_message = message?.message;
+ let detail_message = {...message?.message};

- if (detail_message.message_type === MESSAGE_TYPE.TEXT) {
+ if (detail_message?.message_type === MESSAGE_TYPE.TEXT) {
    detail_message.attachments = [];
  }

  return detail_message;
```
This way, you avoid mutating the original `message` object and also add an extra null check for `detail_message`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->